### PR TITLE
UI: better UI for CommonsForm

### LINKED ISSUE #10: UX FIX: Make fields narrower on Create Commons Form
# User Story

* As an admin or developer
* I can create/edit a new commons more easily
* So that I am less annoyed

### Discussion

The form used on the CreateCommons Page and the Edit Commons page could use some UI tweaks.

Currently the size of the fields scales with the page as it grows and shrinks.  But all of the fields are pretty short (numbers); that space is just wasted.  It might be more reasonable to set a fixed width in characters or pixels, since the maximum size needed would pretty much fit on even the tiniest screens.

The date field is particularly annoying, since the date picker is all the way at the right, so you have to move the mouse a long way to reach it.

image

Default values on the create form, and/or suggested values (or advice on picking a value, brief explanation of what the values mean) might also be helpful.

I suggest looking at the various ways of doing layout available in React Bootstrap (such as `Row` and `Col`;  see links below) and reorganizing the page for convenience:

* https://react-bootstrap.netlify.app/docs/layout/grid
* https://react-bootstrap.netlify.app/docs/forms/overview

You can also include default values as was done by the s23-6pm-3 team---visit their Create Commons page and see what they did:

https://proj-happycows-s23-6pm-3.dokku-12.cs.ucsb.edu/

This is what it looks like when you first load the form; notice how default values are suggested:

image

Here's what the code looks like for that; notice the various uses of  code such as `defaultValue={10000}`

* <https://github.com/ucsb-cs156-s23/proj-happycows-s23-6pm-3/blob/main/frontend/src/main/components/Commons/CommonsForm.js>

One thing that would be nicer is if these default values were all organized, for example, in an object at the 
top of the file, something like:

```js
const defaultValues = {
  startingBalance: 10000,
  cowPrice: 100,
  milkPrice: 20,
  // etc...
} 
```

And then each default value was something like: 
* `defaultValue={defaultValues.startingBalance}` instead of `defaultValue={10000}`
* `defaultValue={defaultValues.cowPrice}` instead of `defaultValue={100}`
* etc...

### Acceptance Criteria

- [x] The field widths have been adjusted in a way that makes the form more usable.
- [x] The date field, in particular, has been adjusted so that the place you click is not so far to the right
- [x] There are suitable default values when you are creating a commons.
- [x] The default value are organized into an easy to read object near the top of the `CommonsForm.js` file.

Optional but helpful:
- [x] The start date defaults to today's date
- [ ] The end date defaults to today's date plus 30 days
- [x] There are explanations (in small font near the field, or in "tooltips" that come up when you hover over a field).

### PR DESCRIPTION
**Pull request by yuanchenmeng**
## Overview
In this PR, I created a better UI design for CommonsForm in bootstrap. Borderlines, default values, and tooltips in form fields are added. Width of form fields is adjusted to adapt to shorter inputs. The date field within the form has been enhanced by fixing its size. 
See Screenshot below.

## Dependencies & Merge Prerequisites
Merge #38 into main first. Then I'll do rebase for some commits. 
This PR will be ready to merge when rebase is done. 

## Screenshots
image

## Tests
Front-end Coverage and mutation tested locally

## Linked Issues
Closes #10

### CODE REVIEW

**Review by iain801 — DISMISSED** [review]
Lgtm

File: .github/workflows/33-frontend-pr-mutation-testing.yml
Comment (dscpsyl) [inline .github/workflows/33-frontend-pr-mutation-testing.yml]: Could we possibly move this change to a different branch? That way, we can keep the commit history clean. Unless there is a reason this change is tied with the UI improvement?

Comment (dscpsyl) [reply .github/workflows/33-frontend-pr-mutation-testing.yml]: Seems like this is a commit for #38. Could we rebase and remove this from this pr? Thanks!

Comment (yuanchenmeng) [reply .github/workflows/33-frontend-pr-mutation-testing.yml]: Yeah. I did exactly same as #38. Otherwise, the frontend mutation testing workflow doesn't work. And I'm not very familiar with git rebase ...

Comment (yuanchenmeng) [reply .github/workflows/33-frontend-pr-mutation-testing.yml]: Or I can wait for #38 merged into main first, then I'll roll back my commit.

Comment (dscpsyl) [reply .github/workflows/33-frontend-pr-mutation-testing.yml]: In this case, I think it's fine since your code can't work without this change. The best thing to do is wait for #38 to be merged first, and then I can help you rebase your commits. On the other hand, if this is merged first, we can also resolve #38 as a duplicate of this PR. Just make sure you write in your PR description that we also changed the workflows.

Comment (yuanchenmeng) [reply .github/workflows/33-frontend-pr-mutation-testing.yml]: Sounds good! I'll update my PR description.

File: frontend/src/main/components/Commons/CommonsForm.js
Comment (pconrad) [inline frontend/src/main/components/Commons/CommonsForm.js]: Starting Balance step should still be 0.01, not 0.001.

It's degradation rate where we want a step value  of 0.001 (in fact, Prof. de Vries now wants `0.0001` instead.  But only for degradation rate.)

File: frontend/src/main/components/Commons/CommonsForm.js
Comment (pconrad) [inline frontend/src/main/components/Commons/CommonsForm.js]: Per Prof. de Vries:

```suggestion
                            step="0.0001"
```

File: frontend/src/tests/components/Commons/CommonsForm.test.js
Comment (pconrad) [inline frontend/src/tests/components/Commons/CommonsForm.test.js]: Do we still need this?  I suspect the answer is no.
```suggestion
```

**Review by pconrad — CHANGES_REQUESTED** [review]
Very nice work overall.  A couple of small issues that we should perhaps address before merging.

**Review by dscpsyl — DISMISSED** [review]
Changes that fix the recommendation. Very nice! Now just resolve the merge conflicts, and we'll be good to go.

**Comment by yuanchenmeng** [comment]
Merge conflicts resolved

**Review by dscpsyl — APPROVED** [review]
Good merge

**Review by pconrad — APPROVED** [review]
LGTM!

**Pull request merged**
